contracts/GenArt721CoreV3.sol

```diff
@@ -817,21 +819,24 @@ contract GenArt721CoreV3 is
     /**
      * @notice Updates script type for project `_projectId`.
      * @param _projectId Project to be updated.
-     * @param _scriptType Libary to be injected by renderer. e.g. "p5js"
-     * @param _scriptTypeVersion Version of library to be injected. e.g. "1.0.0"
+     * @param _scriptTypeAndVersion Script type and version e.g. "p5js@1.0.0",
+     * as bytes32 encoded string.
      */
     function updateProjectScriptType(
         uint256 _projectId,
-        string memory _scriptType,
-        string memory _scriptTypeVersion
+        bytes32 _scriptTypeAndVersion
     )
         external
         onlyUnlocked(_projectId)
         onlyArtistOrAdminACL(_projectId, this.updateProjectScriptType.selector)
     {
         Project storage project = projects[_projectId];
-        project.scriptType = _scriptType;
-        project.scriptTypeVersion = _scriptTypeVersion;
+        // require @ symbol in _scriptTypeAndVersion
+        require(
+            _scriptTypeAndVersion.containsCharacter(AT_CHARACTER_CODE),
+            "must contain @"
+        );
+        project.scriptTypeAndVersion = _scriptTypeAndVersion;
```

Comment:
I wonder if we want to require that it contains _only one_ `@`?

Reply:
Would this potentially conflict with npm's convention? e.g. `@mattheath/p5js-node` is a valid npm package@1.3.7

Reply:
Hrm–that is valid... though don't know we'd ever support an arbitrary NPM package such that this would be relevant. I'm worried if we don't enforce _just one `@`_, that we put ourselves in a weird spot wrt to parsing out the type and version separately if/as we need to in our backend infra. Though based on recent convos with @lyaunzbe , this may be less important as a function of time in the future.

Reply:
true true... Unless we make "allowed qty of @ symbols" an admin configurable qty, I'm slightly inclined to lean on the more permissive side of things for on-chain validation since it is not updatable.  We could always handle any one-off quirks in subgraph. That being said, I don't have very strong opinions on this, and we can work around any one-off quirks different ways (for several reasons, e.g. we overflow 32-byte limit).

Want to make this require only a single `@`?

Reply:
Hrm–yeah my thinking is to just make require a single `@` to have this map to our actual current assumptions–but would love a gut check from @yoshiwarab or @lyaunzbe as well before we ship (or both).

Reply:
i think requiring one `@` is fine here. as mentioned above, npm uses that first `@` for scoping and I do think that's only relevant in the context of npm ultimately

Reply:
here's a specific example also: 
```
https://www.npmjs.com/package/@babylonjs/core 
https://cdnjs.cloudflare.com/ajax/libs/babylonjs/5.20.0/babylon.js
```
even though this dependency does use scoping, it only matters within the context of the npm pkg

Reply:
Yeah I think one @ makes sense to me and will make it a little safer to break the dependency type into type and version components.

Reply:
SGTM! updated 🙏